app/src/main/java/com/example/pygmyhippo/common/UserProcessing/UserProcessing.java

```diff
@@ -0,0 +1,212 @@
+        this.userEmail = email;
+        this.userPhone = phone;
+        this.userPronouns = pronouns;
+        this.userLocation = location;
+        this.deviceID = deviceId;
+        this.accountId = accountId;
+        this.userGeolocation = allowGeo;
+        this.userNotify = allowNotifications;
+        this.context = activity.getApplicationContext();
+        this.imageView = new ImageView(context);
+
+        if (imagePath == null) {
+            this.avatarPath = imageUrl + userName;
+            loadAndUpload();
+            System.out.println("generate avatar..");
+        } else {
+            System.out.println("Creating...");
+            Account newUser = new Account(
+                    userEmail + deviceID,
+                    userName,
+                    userPronouns,
+                    userPhone,
+                    userEmail,
+                    deviceID,
+                    imagePath,
+                    null,
+                    userLocation,
+                    userNotify,
+                    userGeolocation
+            );}
+    }
+
+    /** Code is from Chatgpt for uploading from Picasso's cache to firebase
+     * Prompts: Access the files stored in cache by picasso, I need (it) to be from from cache because firebase cannot load from a url
+     * Accessed November 2, 2024
+     * @Author: Chatgpt, implemented and Edited by Jennifer
+     *
+     * */
+    public void loadAndUpload() {
+        // Create a Target to handle the bitmap,
+        Target target = new Target() {
+            @Override
+            public void onBitmapLoaded(Bitmap bitmap, Picasso.LoadedFrom from) {
+                // Upload the bitmap to Firebase
+                try {
+                    uploadAvatar(bitmap);
+                    System.out.println("Setting avatar");
+                } catch (FileNotFoundException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+
+            @Override
+            public void onBitmapFailed(Exception e, Drawable errorDrawable) {
+                // Handle loading error
+                System.out.println("Bitmap failed" + e);
+            }
+
+            @Override
+            public void onPrepareLoad(Drawable placeHolderDrawable) {
+                // Optional: Prepare for loading
+            }
+        };
+
+        // Load the image into the Target
+        Picasso.get()
+                .load(avatarPath)
+                .into(target);
+    }
+
+    /** Code is to upload generated avatar to firebase, first two lines are final bit of code from chatgpt, the rest is from the fisebase docs on uploading files
+     * https://firebase.google.com/docs/storage/android/upload-files
+     * Accessed November 2, 2024
+     * @Author: Jennifer and Firebase documentation
+     * @version 1.0
+     * */
+    private void uploadAvatar(Bitmap bitmap) throws FileNotFoundException {
+        // Convert bitmap to byte array for upload
```

Comment:
Are we going to be using bitmaps for image uploads? I am open to this idea as long as we all use bitmaps for images. I guess just make sure everyone knows to use bitmaps instead of firebase store.

Reply:
The user selected avatar uploads from a string, but the generated Avatar uploads from a byte array, the bitmap is just one step of the process. It's not ideal, but it's been extremely difficult to get generated avatars to upload at all, even moving the byteArray out of userProcessing will break it unless you move all the logic used to download the avatar as well.

Reply:
I have no problem with it, it's just something that we should all be doing the same. So moving forward let's say all images are stored in bitmaps in the database.

Reply:
I haven't had any luck with converting regular images into bitmaps, even small ones, it creates a null exception pointer. The image path is correct, as it's what is used to upload via string, I've tried adding a time delay in case it takes more time to process which hasn't worked either.